### ISSUE TITLE
"OSError: -9" saving a multi-layer TIFF via libtiff

### ISSUE THREAD

**Issue by ndjman7** [python-pillow/Pillow#5319]
### What did you do?
I did a version up. (5.2.0 -> 6.2.0)
It was the same at 8.1.2.

This is similar to this issue.
https://stackoverflow.com/questions/64683811/how-to-debug-oserror-9-saving-a-tiff-via-libtiff-and-pillow

### What did you expect to happen?
I thought dividing the multi-layer TIFF files would work the same way.

### What actually happened?
raise Error `OSError -9`

```python
...
# example
with Image.open(document_file) as tif_images:
    result = []

    for tif_image in ImageSequence.Iterator(tif_images):
        png_image_bytes_io = io.BytesIO()
        tif_image.save(png_image_bytes_io, format='png')  # <- raise Error
        result.append(png_image_bytes_io)
...
```

### What are your OS, Python and Pillow versions?

* OS: macOS Catalina 10.15.7
* Python: 3.6.9
* Pillow: 6.2.0

![image](https://user-images.githubusercontent.com/15621458/110486540-14a98780-8130-11eb-9983-08ece0e6f474.png)

```shell
Traceback:

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/django/views/generic/base.py" in dispatch
  97.         return handler(request, *args, **kwargs)

File "/Users/ra/Projects/inapi/unsecured/views/upload_fax_document.py" in post
  105.             png_images = convert_tif_file_to_png_files(document_file.file)

File "/Users/ra/Projects/inapi/unsecured/views/upload_fax_document.py" in convert_tif_file_to_png_files
  61.             tif_image.save(png_image_bytes_io, format='png')

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/PIL/Image.py" in save
  2050.         self._ensure_mutable()

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/PIL/Image.py" in _ensure_mutable
  640.             self._copy()

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/PIL/Image.py" in _copy
  633.         self.load()

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/PIL/TiffImagePlugin.py" in load
  1097.             return self._load_libtiff()

File "/Users/ra/.pyenv/versions/inapi/lib/python3.6/site-packages/PIL/TiffImagePlugin.py" in _load_libtiff
  1209.             raise IOError(err)

Exception Type: OSError at /unsecured/upload_fax_document/
Exception Value: -9
```

**Comment by radarhere** [comment]
Ok, so you're trying to iterate through the frames of a TIFF file and save those frames individually, right?

Could we get a copy of the starting image?

**Comment by ndjman7** [comment]
@radarhere I replied late.

[pando.tif.zip](https://github.com/python-pillow/Pillow/files/6120715/pando.tif.zip)

**Comment by ndjman7** [comment]
@radarhere Did you check by any chance?

**Comment by radarhere** [comment]
I can reproduce the problem with just
```python
from PIL import Image
with Image.open('pando.tif') as im:
    im.save('out.tif')
```
Debugging, I find that it is failing at https://github.com/python-pillow/Pillow/blob/759c518420c70397801e571bd523ec6e4735bdab/src/libImaging/TiffDecode.c#L339-L343

Pillow has determined `rows_per_strip` is 4294967295. That's large.

If I run the command line tool tiffinfo on the file, the first part of its report is
```
TIFF Directory at offset 0x14dd6 (85462)
  Image Width: 1728 Image Length: 2344
  Resolution: 204, 196 pixels/inch
  Bits/Sample: 1
  Compression Scheme: CCITT Group 3
  Photometric Interpretation: min-is-white
  FillOrder: lsb-to-msb
  Orientation: row 0 top, col 0 lhs
  Samples/Pixel: 1
  Rows/Strip: (infinite)
```
Infinite is also pretty large.

**Comment by ndjman7** [comment]
@radarhere Yes, you can. Thank you for the solution.

**Comment by radarhere** [comment]
This has been resolved instead by #5364. The fix should be a part of 8.2.0 when it is released on April 1.
